nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,525 @@
+                    "work in the bytecode submission setting should also be contained in these JAR files.")
+            .dependsOn(CONNECTION_SETTINGS, YAML_SETTINGS)
+            .defaultValue(null)
+            .identifiesExternalResource(ResourceCardinality.MULTIPLE, ResourceType.FILE, ResourceType.DIRECTORY, ResourceType.URL)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .dynamicallyModifiesClasspath(true)
+            .build();
+
+    public static final PropertyDescriptor EXTENSION_CLASSES = new PropertyDescriptor.Builder()
+            .name("extension-classes")
+            .displayName("Extension Classes")
+            .addValidator(Validator.VALID)
+            .description("A comma-separated list of fully qualified Java class names that correspond to classes to implement. This " +
+                    "is useful for services such as JanusGraph that need specific serialization classes. " +
+                    "This configuration property has no effect unless a value for the Extension JAR field is " +
+                    "also provided.")
+            .dependsOn(EXTRA_RESOURCE)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .dependsOn(CONNECTION_SETTINGS, YAML_SETTINGS)
+            .required(false)
+            .build();
+
+    public static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("ssl-context-service")
+            .displayName("SSL Context Service")
+            .description("The SSL Context Service used to provide client certificate information for TLS/SSL "
+                    + "connections.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    public static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            SUBMISSION_TYPE,
+            CONNECTION_SETTINGS,
+            REMOTE_OBJECTS_FILE,
+            EXTRA_RESOURCE,
+            EXTENSION_CLASSES,
+            CONTACT_POINTS,
+            PORT,
+            PATH,
+            TRAVERSAL_SOURCE_NAME,
+            USER_NAME,
+            PASSWORD,
+            SSL_CONTEXT_SERVICE
+    ));
+
+//    private ScriptEngineManager MANAGER = new ScriptEngineManager();
+    private GroovyShell groovyShell;
+//    private ScriptEngine engine;
+    private Map<String, Script> compiledCode;
+    protected Cluster cluster;
+    private String traversalSourceName;
+    private GraphTraversalSource traversalSource;
+    private boolean scriptSubmission = true;
+    boolean usesSSL;
+    protected String transitUrl;
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException {
+        loadClasses(context);
+        GroovyClassLoader loader = new GroovyClassLoader(this.getClass().getClassLoader());
+        groovyShell = new GroovyShell(loader, new Binding());
```

Comment:
@mattyb149 I think [this](https://docs.groovy-lang.org/latest/html/api/groovy/lang/GroovyShell.html#%3Cinit%3E(java.lang.ClassLoader)) is the safer constructor.

Reply:
Updated to this constructor.